pkg/machines/reducers.es6

```diff
@@ -147,7 +147,7 @@ function vms(state, action) {
                 : getFirstIndexOfVm(state, 'name', action.payload.name, connectionName);
             if (index < 0) {
                 logDebug(`VM_ACTION_FAILED reducer(name='${action.payload.name}', connectionName='${connectionName}') not found, skipping`);
-                return ;
+                return state; // VM is (already) missing, do nothing, the failure was at least logged
```

Comment:
This seems like a candidate for a unit tests, especially if we expect other providers to be written.